Fails to build with libglib2.0-dev:2.62.4

On Ubuntu 19.10 I got errors like this:
```
libdrakvuf.h:187:5: error: ‘GTimeVal’ is deprecated: Use 'GDateTime' instead [-Werror=deprecated-declarations]                                                                                
  187 |     GTimeVal timestamp;                                                                                                                                                               
      |     ^~~~~~~~                                                                           
```

There is no such error with `2.60.4`.

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -138,11 +138,13 @@ AM_LDFLAGS += $(SANITIZE_LDFLAGS)
 endif
 
 # Note that -pg is incompatible with HARDENING
-if DEBUG
+if !DEBUG
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
+else
 AM_CFLAGS += -DDRAKVUF_DEBUG -Werror -Wall -Wextra -g -ggdb3
-AM_CXXFLAGS += -Wno-c99-designator -Wno-unknown-warning-option
 AM_CXXFLAGS += -DDRAKVUF_DEBUG -Werror -Wall -Wextra -g -ggdb3
-AM_CXXFLAGS += -ferror-limit=0
+AM_CXXFLAGS += -ferror-limit=0 -Wno-unknown-warning-option
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
 if !HARDENING
 AM_CFLAGS += -pg
 AM_CXXFLAGS += -pg

--- a/src/libdrakvuf/libdrakvuf.h
+++ b/src/libdrakvuf/libdrakvuf.h
@@ -184,7 +184,7 @@ typedef struct drakvuf_trap drakvuf_trap_t;
 
 typedef struct drakvuf_trap_info
 {
-    GTimeVal timestamp;
+    gint64 timestamp;
     unsigned int vcpu;
     uint16_t altp2m_idx;
     proc_data_t proc_data ; /* Current executing process data */
@@ -617,12 +617,11 @@ typedef enum
 
 // Printf helpers for timestamp.
 #define FORMAT_TIMEVAL "%" PRId64 ".%06" PRId64
-#define UNPACK_TIMEVAL(t) (t).tv_sec, (t).tv_usec
+#define UNPACK_TIMEVAL(t) (t/G_USEC_PER_SEC), (t - t/G_USEC_PER_SEC)
 
 #define eprint_current_time(...) \
     do { \
-        GTimeVal current_time; \
-        g_get_current_time (&current_time); \
+        gint64 current_time = g_get_real_time(); \
         fprintf(stderr, FORMAT_TIMEVAL " ", UNPACK_TIMEVAL(current_time)); \
     } while (0)
 

--- a/src/libdrakvuf/vmi.c
+++ b/src/libdrakvuf/vmi.c
@@ -227,7 +227,7 @@ event_response_t post_mem_cb(vmi_instance_t vmi, vmi_event_t* event)
             if (!trap_info.proc_data.tid)
                 drakvuf_get_current_thread_id(drakvuf, &trap_info, &trap_info.proc_data.tid);
 
-            g_get_current_time(&trap_info.timestamp);
+            trap_info.timestamp = g_get_real_time();
 
             rsp |= trap->cb(drakvuf, &trap_info);
         }
@@ -364,17 +364,14 @@ event_response_t pre_mem_cb(vmi_instance_t vmi, vmi_event_t* event)
                 (event->mem_event.out_access & VMI_MEMACCESS_X) ? 'x' : '-'
                );
 
-    GTimeVal timestamp;
-    g_get_current_time(&timestamp);
-
     proc_data_priv_t proc_data;
     memset(&proc_data, 0, sizeof(proc_data_priv_t));
     drakvuf_trap_info_t trap_info;
     memset(&trap_info, 0, sizeof(drakvuf_trap_info_t));
 
+    trap_info.timestamp = g_get_real_time();
     trap_info.regs = event->x86_regs;
     trap_info.vcpu = event->vcpu_id;
-    trap_info.timestamp = timestamp;
     trap_info.trap_pa = pa;
 
     drakvuf_get_current_process_data( drakvuf, &trap_info, &proc_data );
@@ -543,17 +540,14 @@ event_response_t int3_cb(vmi_instance_t vmi, vmi_event_t* event)
     else
         event->interrupt_event.reinject = 0;
 
-    GTimeVal timestamp;
-    g_get_current_time(&timestamp);
-
     proc_data_priv_t proc_data;
     memset(&proc_data, 0, sizeof(proc_data_priv_t));
     drakvuf_trap_info_t trap_info;
     memset(&trap_info, 0, sizeof(drakvuf_trap_info_t));
 
+    trap_info.timestamp = g_get_real_time();
     trap_info.regs = event->x86_regs;
     trap_info.vcpu = event->vcpu_id;
-    trap_info.timestamp = timestamp;
     trap_info.trap_pa = pa;
 
     drakvuf_get_current_process_data( drakvuf, &trap_info, &proc_data );
@@ -611,17 +605,14 @@ event_response_t cr3_cb(vmi_instance_t vmi, vmi_event_t* event)
 
     event->x86_regs->cr3 = event->reg_event.value;
 
-    GTimeVal timestamp;
-    g_get_current_time(&timestamp);
-
     proc_data_priv_t proc_data;
     memset(&proc_data, 0, sizeof(proc_data_priv_t));
     drakvuf_trap_info_t trap_info;
     memset(&trap_info, 0, sizeof(drakvuf_trap_info_t));
 
+    trap_info.timestamp = g_get_real_time();
     trap_info.regs = event->x86_regs;
     trap_info.vcpu = event->vcpu_id;
-    trap_info.timestamp = timestamp;
 
     drakvuf_get_current_process_data( drakvuf, &trap_info, &proc_data );
 
@@ -664,17 +655,14 @@ event_response_t debug_cb(vmi_instance_t vmi, vmi_event_t* event)
                 event->debug_event.gla, event->debug_event.insn_length);
 #endif
 
-    GTimeVal timestamp;
-    g_get_current_time(&timestamp);
-
     proc_data_priv_t proc_data;
     memset(&proc_data, 0, sizeof(proc_data_priv_t));
     drakvuf_trap_info_t trap_info;
     memset(&trap_info, 0, sizeof(drakvuf_trap_info_t));
 
+    trap_info.timestamp = g_get_real_time();
     trap_info.regs = event->x86_regs;
     trap_info.vcpu = event->vcpu_id;
-    trap_info.timestamp = timestamp;
     trap_info.debug = &event->debug_event;
 
     drakvuf_get_current_process_data( drakvuf, &trap_info, &proc_data );
@@ -719,17 +707,14 @@ event_response_t cpuid_cb(vmi_instance_t vmi, vmi_event_t* event)
 
     reg_t rip = event->x86_regs->rip;
 
-    GTimeVal timestamp;
-    g_get_current_time(&timestamp);
-
     proc_data_priv_t proc_data;
     memset(&proc_data, 0, sizeof(proc_data_priv_t));
     drakvuf_trap_info_t trap_info;
     memset(&trap_info, 0, sizeof(drakvuf_trap_info_t));
 
+    trap_info.timestamp = g_get_real_time();
     trap_info.regs = event->x86_regs;
     trap_info.vcpu = event->vcpu_id;
-    trap_info.timestamp = timestamp;
     trap_info.cpuid = &event->cpuid_event;
 
     drakvuf_get_current_process_data( drakvuf, &trap_info, &proc_data );

--- a/src/libinjector/linux_injector.c
+++ b/src/libinjector/linux_injector.c
@@ -710,8 +710,7 @@ static bool load_file_to_memory(addr_t* output, size_t* size, const char* file)
 
 static void print_injection_info(output_format_t format, const char* file, injector_t injector)
 {
-    GTimeVal t;
-    g_get_current_time(&t);
+    gint64 t = g_get_real_time();
     char* arguments = "";
 
     for (int i=0; i<injector->args_count; i++)

--- a/src/libinjector/win_injector.c
+++ b/src/libinjector/win_injector.c
@@ -1248,8 +1248,7 @@ static bool load_file_to_memory(addr_t* output, size_t* size, const char* file)
 
 static void print_injection_info(output_format_t format, const char* file, injector_t injector)
 {
-    GTimeVal t;
-    g_get_current_time(&t);
+    gint64 t = g_get_real_time();
 
     char* process_name = NULL;
     char* arguments = NULL;

--- a/src/libusermode/Makefile.am
+++ b/src/libusermode/Makefile.am
@@ -117,6 +117,7 @@ AM_LDFLAGS += $(VMI_LIBS)
 AM_LDFLAGS += $(JSONC_LIBS)
 
 AM_CXXFLAGS = $(CXXFLAGS)
+
 if HARDENING
 AM_CXXFLAGS += $(HARDEN_CFLAGS) -DHARDENING
 endif
@@ -126,10 +127,12 @@ AM_CXXFLAGS += $(SANITIZE_CFLAGS)
 AM_LDFLAGS += $(SANITIZE_LDFLAGS)
 endif
 
-if DEBUG
+if !DEBUG
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
+else
 AM_CXXFLAGS += -DDRAKVUF_DEBUG -Werror -Wall -Wextra -g -ggdb3
-AM_CXXFLAGS += -Wno-unused-parameter
-AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list -Wno-unknown-warning-option
+AM_CXXFLAGS += -Wno-unused-parameter -Wno-unknown-warning-option
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
 AM_CXXFLAGS += -Wcast-qual -Wcast-align -Wstrict-aliasing \
                -Wpointer-arith -Winit-self -Wshadow \
                -Wredundant-decls -Wfloat-equal -Wundef \

--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -272,6 +272,7 @@ AM_LDFLAGS += $(VMI_LIBS)
 AM_LDFLAGS += $(JSONC_LIBS)
 
 AM_CXXFLAGS = $(CXXFLAGS)
+
 if HARDENING
 AM_CXXFLAGS += $(HARDEN_CFLAGS) -DHARDENING
 AM_LDFLAGS += $(HARDEN_LDFLAGS)
@@ -282,15 +283,17 @@ AM_CXXFLAGS += $(SANITIZE_CFLAGS)
 AM_LDFLAGS += $(SANITIZE_LDFLAGS)
 endif
 
-if DEBUG
+if !DEBUG
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
+else
 AM_CXXFLAGS += -DDRAKVUF_DEBUG -Werror -Wall -Wextra -g -ggdb3
 AM_CXXFLAGS += -ferror-limit=0
 AM_CXXFLAGS += -Wno-unused-parameter -Wno-unknown-warning-option
+AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list
 AM_CXXFLAGS += -Wcast-qual -Wcast-align -Wstrict-aliasing \
                -Wpointer-arith -Winit-self -Wshadow \
                -Wredundant-decls -Wfloat-equal -Wundef \
                -Wvla
-AM_CXXFLAGS += -Wno-c99-designator -Wno-reorder-init-list -Wno-unknown-warning-option
 # Note that -pg is incompatible with HARDENING
 if !HARDENING
 AM_CXXFLAGS += -pg

--- a/src/plugins/output_format.h
+++ b/src/plugins/output_format.h
@@ -121,12 +121,6 @@ auto keyval(const char* key, const Value& value)
     return std::make_pair(key, value);
 }
 
-std::ostream& operator<<(std::ostream& os, const GTimeVal& t)
-{
-    os << t.tv_sec << "." << std::to_string(t.tv_usec).substr(0, 6);
-    return os;
-}
-
 namespace fmt
 {
 

--- a/src/plugins/procmon/procmon.cpp
+++ b/src/plugins/procmon/procmon.cpp
@@ -784,8 +784,7 @@ static void process_visitor(drakvuf_t drakvuf, addr_t process, void* visitor_ctx
         return;
     }
 
-    GTimeVal t;
-    g_get_current_time(&t);
+    gint64 t = g_get_real_time();
 
     switch (ctx->format)
     {

